lwt/protocol_lwt.ml

```diff
@@ -39,16 +39,19 @@ module M = struct
     let fd = Lwt_unix.socket Lwt_unix.PF_UNIX Lwt_unix.SOCK_STREAM 0 in
     let rec loop () =
       Lwt.catch (fun () ->
-          Lwt_unix.connect fd sockaddr >>= fun () ->
-          let ic = Lwt_io.of_fd ~close:(fun () -> Lwt_unix.close fd) ~mode:Lwt_io.input fd in
-          let oc = Lwt_io.of_fd ~close:(fun () -> return ()) ~mode:Lwt_io.output fd in
-          return (ic, oc)
+          Lwt_unix.connect fd sockaddr
         ) (function
           | Unix.Unix_error((Unix.ECONNREFUSED | Unix.ECONNABORTED | Unix.ENOENT), _, _) ->
             Lwt_unix.sleep 5. >>= fun () ->
             loop ()
-          | e -> fail e
-        ) in
+          | e ->
+            Lwt_unix.close fd >>= fail e
+        )
+      >>= fun () ->
+      let ic = Lwt_io.of_fd ~close:(fun () -> Lwt_unix.close fd) ~mode:Lwt_io.input fd in
```

Comment:
Worth mentioning that `(fun () -> Lwt_unix.close fd)` is the default value for `?close`, but I'd keep it there for clarity

Reply:
I think it's fine anyways